### DISCUSSION TITLE
Deprecate a block

### DISCUSSION THREAD

**Discussion by Suriteka** [WordPress/gutenberg#37744] [Q&A] [answered]
Hello 👋 

First of all, Thanks a lot for Gutenberg and this awesome editor.

So, I had a custom block, which had this as `save.js` :

```js
<summary {...blockProps} >
    <InnerBlocks.Content />
</details>
```

And that is now

```js
return <RichText.Content {...blockProps} tagName="summary" value={attributes.content}/>
```
and with a new attribute in `attribute.js`

```js
const attributes = {
  content: {
    "type": "string",
    "source": 'html',
    "selector": 'summary'
  }
}
```

I'm now trying to create the deprecation of the old save.
We currently have in documentation the example to do this the other way (from Text to InnerContent), but how can I get the content of an innerBlock to pass the data to my `RichText.Content`?

Do you have an idea?
Thanks a lot,

**Comment by talldan** [comment, marked as answer]
That's a good question. I don't think there's a reliable way to do this. The problem is that `RichText` is naturally less capable than `InnerBlocks`, so you won't be able to reliably convert from inner blocks to rich text without some data loss.

The amount of data loss really depends on the makeup of your inner blocks in the old version of your block. 

If the block was completely freeform and any block was allowed to be added as an inner block, then it becomes a bit tricky. Your best bet is to use `migrate` in the deprecation. You get access to the inner blocks as the second param to that function. Your challenge will be to assemble those inner blocks into a `content` string. You could probably start by using the `wp.blocks.serialize` function to convert those blocks to HTML, but that HTML will contain stuff like the block delimiters and HTML tags. You'll then need to strip some of that out. You'll also lose any dynamic blocks since they can't be rendered by the client.

If you had `allowedBlocks` specified in the old version of your block, and those known blocks also had `content` attributes, then your best option would be to assemble your `content` by reading those attributes from the inner blocks provided by the migrate function, and assemble that into a content string.

**Reply by Suriteka** [reply]
Thanks a lot for your answer!
